db/db.py

```diff
@@ -271,27 +281,78 @@ class Job(Base):
     timeout_seconds = Column(Integer, nullable=False)
     email_notifications = Column(JSON, nullable=False)
     workspace_id = Column(String, ForeignKey("workspaces.id"), nullable=False)
-    workspace = relationship(Workspace)
     new_cluster = Column(JSON)
+    existing_cluster_id = Column(String)
     schedule_quartz_cron_expression = Column(String)
     schedule_timezone_id = Column(String)
     task_type = Column(String)
-    notebook_path = Column(String)
-    notebook_revision_timestamp = Column(Integer)
+    task_parameters = Column(JSON)
     max_concurrent_runs = Column(Integer)
 
+    workspace = relationship(Workspace)
+    jobruns = relationship("JobRun",
+                           back_populates='job',
+                           order_by="JobRun.start_time")
+
+    def runs(self, last=None, since_days=None):
+        runs = self.jobruns
+
+        if since_days is not None:
+            runs = [run for run in runs if run.start_time >= since(since_days)]
+
+        if last is not None:
+            runs = runs[-last:]
+
+        return runs
+
+    def num_runs(self, last=None, since_days=None):
+        return len(self.runs(last, since_days))
+
+    def duration(self, last=None, since_days=None):
+        return sum([run.duration() for run in self.runs(last, since_days)])
+
+    def dbu(self, last=None, since_days=None):
+        return sum([run.dbu() for run in self.runs(last, since_days)])
+
+    def cost(self, price_config, last=None, since_days=None):
+        return sum([run.cost(price_config)
+                    for run in self.runs(last, since_days)])
+
+    def run_df(self, price_config=None, last=None, since_days=None):
```

Comment:
The goal of the function is to return the run data as a pandas Dataframe, right? If yes, this name is not very descriptive. Maybe something like: `toDF`, `to_dataframe` or `to_pandas_dataframe` would describe the intent better.

Reply:
This name follows the convention we used in cluster table's `state_df` method.

Reply:
Method merged into `runs` method with a conversion parameter added.